test/test_torch.py

```diff
@@ -9554,6 +9554,28 @@ def test_cdist_norm_batch(self, device):
                         expected = brute_cdist(x, y, p=p)
                         self.assertTrue(torch.allclose(expected, actual))
 
+    def test_multinomial_unnormalized(self, device):
+        for dtype in [torch.float, torch.double]:
+            part_size = 15
+            num_samples = 2 * part_size
+            # ensure [0.01 - 1.01] range
+            large_part = torch.rand(part_size, dtype=dtype, device=device) + 0.01
+            small_part = torch.rand(part_size, dtype=dtype, device=device) + 0.01
+            large_part = large_part * 1e15
+            small_part = small_part * 1e-15
+            dist = torch.cat((large_part, small_part), 0)
+            result = torch.multinomial(dist, num_samples, replacement=False)
+            actual = torch.unique(result, sorted=True)
+            expected = torch.from_numpy(np.arange(0, num_samples)).to(device)
+            self.assertEqual(actual, expected, "non-unique indicies")
```

Comment:
"non-unique indices" is not very descriptive (i.e. what has non-unique indices? Why is it a problem that there are non-unique indices?), could we improve it?

Reply:
Also, I'm not sure that `message` is the third argument to `self.assertEqual` (I think precision is the third argument). Please pass it as a kwarg.